services/wfm/src/main/java/org/openkilda/wfm/topology/utils/KibanaLogManager.java

```diff
@@ -0,0 +1,135 @@
+/* Copyright 2018 Telstra Open Source
+ *
+ *   Licensed under the Apache License, Version 2.0 (the "License");
+ *   you may not use this file except in compliance with the License.
+ *   You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *   distributed under the License is distributed on an "AS IS" BASIS,
+ *   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *   See the License for the specific language governing permissions and
+ *   limitations under the License.
+ */
+
+package org.openkilda.wfm.topology.utils;
+
+import static org.apache.log4j.Level.ERROR;
+import static org.apache.log4j.Level.INFO;
+import static org.apache.log4j.Level.WARN;
+
+import org.openkilda.messaging.info.discovery.NetworkDumpPortData;
+import org.openkilda.messaging.info.discovery.NetworkDumpSwitchData;
+import org.openkilda.messaging.info.event.IslChangeType;
+import org.openkilda.messaging.info.event.SwitchState;
+import org.openkilda.messaging.model.SwitchId;
+
+import com.google.common.collect.ImmutableMap;
+import org.apache.log4j.Level;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.slf4j.MDC;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.annotation.Nullable;
+
+public class KibanaLogManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(KibanaLogManager.class);
+    /**
+     * The Kibana dashboard tag.
+     */
+    private static final String TAG = "KIBANA_DASHBOARD_TAG";
+
+    public void vfmSync(NetworkDumpSwitchData switchData) {
+        Map<Object, Object> data = ImmutableMap.of(TAG, "Event/WFM Sync", "switch", switchData);
+        proceed("Switch sync", data, INFO);
+    }
+
+    public void vfmSync(NetworkDumpPortData portData) {
+        Map<Object, Object> data = ImmutableMap.of(TAG, "Event/WFM Sync", "switch", portData);
+        proceed("Port sync", data, INFO);
+    }
+
+    public void switchDiscovery(SwitchId switchId, SwitchState state) {
+        Map<Object, Object> data = ImmutableMap.of(TAG, "DISCO: Switch Event", "switch", switchId, "state", state);
+        proceed("Switch event", data, INFO);
+    }
+
+    public void switchEvent(SwitchState state) {
+        Map<Object, Object> data = ImmutableMap.of(TAG, "SWITCH Event", "state", state);
+        proceed("Ignoring state", data, WARN);
+    }
+
+    /**
+     *  Port discovery log.
+     *
+     * @param switchId a switch id.
+     * @param portId a port id.
+     * @param state a port state.
+     */
+    public void portDiscovery(SwitchId switchId, int portId, String state) {
+        Map<Object, Object> data =
+                ImmutableMap.of(TAG, "DISCO: Port Event", "switch", switchId, "port", portId, "state", state);
+        proceed("Port discovery", data, INFO);
+    }
+
+    public void portEvent(String state) {
```

Comment:
Where is the switchId and port number? How you going to report port event without any reference to port triggered this event?